--- ./src/eventlistitem.cpp.orig	2022-02-27 17:05:06.870547741 +0000
+++ ./src/eventlistitem.cpp	2022-02-27 17:05:15.056515593 +0000
@@ -68,7 +68,7 @@ QString EventListItem::getstring() const
             //: Text shown on bookmark markers in the main window marker list
             .arg(QCoreApplication::translate("eventlist", "BOOKMARK"));
 
-  return label + QString().sprintf("<br>%02d:%02d:%02d.%03d<br>%d (%c)",
+  return label + QString().asprintf("<br>%02d:%02d:%02d.%03d<br>%d (%c)",
                            int(pts/(3600*90000)),
                            int(pts/(60*90000))%60,
                            int(pts/90000)%60,
