diff -upr dvbcut-0.7.3.orig/configure.ac dvbcut-0.7.3/configure.ac
--- dvbcut-0.7.3.orig/configure.ac	2022-02-27 23:39:19.579651681 +0000
+++ dvbcut-0.7.3/configure.ac	2022-02-27 23:32:11.292333787 +0000
@@ -54,12 +54,35 @@ done
 
 dnl Checks for libraries.
 
-FFMPEG_LIBS='-lavformat -lavcodec -lavutil'
 AC_SUBST(FFMPEG_LIBS)
 
-AC_CHECK_LIB(swscale, main,
-  [AC_DEFINE(HAVE_LIB_SWSCALE, 1, [Define this if you have libswscale.])
-  FFMPEG_LIBS="$FFMPEG_LIBS -lswscale"])
+PKG_CHECK_MODULES(avformat, libavformat,
+  [HAVE_FFMPEG="yes"],
+  [AC_MSG_ERROR([libavformat library not found])])
+
+PKG_CHECK_MODULES(avcodec, libavcodec,
+  [HAVE_FFMPEG="yes"],
+  [AC_MSG_ERROR([libavcodec library not found])])
+
+PKG_CHECK_MODULES(avutil, libavutil,
+  [HAVE_FFMPEG="yes"],
+  [AC_MSG_ERROR([libavutil library not found])])
+
+PKG_CHECK_MODULES(swscale, libavutil,
+  [HAVE_SWSCALE="yes"],
+  [AC_MSG_ERROR([libswscale library not found])])
+
+if test "x$HAVE_FFMPEG" = "xyes"; then
+   FFMPEG_LIBS="`$PKG_CONFIG --libs libavformat libavcodec libavutil`"
+   CPPFLAGS="$CPPFLAGS `$PKG_CONFIG --cflags libavformat libavcodec libavutil`"
+   if test "x$HAVE_SWSCALE" = "xyes"; then
+      AC_DEFINE(HAVE_LIB_SWSCALE, 1, [Define this if you have libswscale.])
+      FFMPEG_LIBS="$FFMPEG_LIBS `$PKG_CONFIG --libs libswscale`"
+      CPPFLAGS="$CPPFLAGS `$PKG_CONFIG --cflags libswscale`"
+   fi
+else
+   AC_MSG_ERROR([cannot find ffmpeg libraries])
+fi
 
 AC_CHECK_LIB(mad, mad_decoder_init,
   [AC_DEFINE(HAVE_LIB_MAD, 1, [Define this if you have libmad.])
