diff -ur dvbcut.orig/src/lavfmuxer.cpp dvbcut/src/lavfmuxer.cpp
--- dvbcut.orig/src/lavfmuxer.cpp	2012-08-31 23:28:42.000000000 +1000
+++ dvbcut/src/lavfmuxer.cpp	2012-09-01 12:12:04.472248687 +1000
@@ -43,9 +43,16 @@
   if (!avfc)
     return;
 
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(53, 35, 0)
+// todo: what here ?
+//  maybe: AVFormatContext::audio_preload but no direct access.
+//    AVOptions
+//    iformat
+#else
   avfc->preload= (int)(.5*AV_TIME_BASE);
-  avfc->max_delay= (int)(.7*AV_TIME_BASE);
   avfc->mux_rate=10080000;
+#endif
+  avfc->max_delay= (int)(.7*AV_TIME_BASE);
 
   avfc->oformat=fmt;
   strncpy(avfc->filename, filename, sizeof(avfc->filename));
@@ -114,19 +121,36 @@
       }
     }
 
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(53, 35, 0)
+  // error: 'av_set_parameters' was not declared in this scope
+  if (!(fmt->flags & AVFMT_NOFILE)&&(avio_open(&avfc->pb, filename, AVIO_FLAG_WRITE) < 0)) {
+#else
   if ((av_set_parameters(avfc, NULL) < 0) || (!(fmt->flags & AVFMT_NOFILE)&&(url_fopen(&avfc->pb, filename, URL_WRONLY) < 0))) {
+#endif  
     av_free(avfc);
     avfc=0;
     return;
     }
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(53, 35, 0)
+// todo: what here ?
+//  maybe: AVFormatContext::audio_preload but no direct access.
+//    AVOptions
+//    iformat
+#else
   avfc->preload= (int)(.5*AV_TIME_BASE);
-  avfc->max_delay= (int)(.7*AV_TIME_BASE);
   avfc->mux_rate=10080000;
+#endif
+  avfc->max_delay= (int)(.7*AV_TIME_BASE);
 
-
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(53, 35, 0)
+  av_dump_format(avfc, 0, filename, 1);
+  fileopened=true;
+  avformat_write_header(avfc, NULL);
+#else
   dump_format(avfc, 0, filename, 1);
   fileopened=true;
   av_write_header(avfc);
+#endif
   }
 
 
@@ -136,7 +160,9 @@
     if (fileopened) {
       av_write_trailer(avfc);
       if (!(fmt->flags & AVFMT_NOFILE))
-#if LIBAVFORMAT_VERSION_INT >= ((52<<16)+(0<<8)+0)
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(53, 35, 0)
+        avio_close(avfc->pb);
+#elif LIBAVFORMAT_VERSION_INT >= ((52<<16)+(0<<8)+0)
         url_fclose(avfc->pb);
 #else
         url_fclose(&avfc->pb);
diff -ur dvbcut.orig/src/mpgfile.cpp dvbcut/src/mpgfile.cpp
--- dvbcut.orig/src/mpgfile.cpp	2012-09-01 12:26:19.000000000 +1000
+++ dvbcut/src/mpgfile.cpp	2012-09-01 12:49:57.133541130 +1000
@@ -733,8 +733,11 @@
       f->pts=idx[idx.indexnr(start+p)].getpts()-startpts;
       f->coded_picture_number=f->display_picture_number=p;
       f->key_frame=(p==0)?1:0;
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(53, 35, 0)      
+      f->pict_type=(p==0)?AV_PICTURE_TYPE_I : AV_PICTURE_TYPE_P;
+#else
       f->pict_type=(p==0)?FF_I_TYPE:FF_P_TYPE;
-
+#endif
       out = avcodec_encode_video(avcc, buf,
                                  m2v.getsize(), f);
 
Only in dvbcut/src: mpgfile.cpp~
