--- dvbcut-svn179.orig/src/lavfmuxer.cpp	2014-10-20 19:36:01.108711000 +1100
+++ dvbcut-svn179/src/lavfmuxer.cpp	2014-10-20 21:35:28.682373468 +1100
@@ -60,7 +60,8 @@
   int id=0;
 
   st[VIDEOSTREAM].stream_index=id;
-  AVStream *s=st[VIDEOSTREAM].avstr=av_new_stream(avfc,id++);
+  AVStream *s=st[VIDEOSTREAM].avstr=avformat_new_stream(avfc, NULL);
+  s->id = id++;
   strpres[VIDEOSTREAM]=true;
   av_free(s->codec);
   mpg.setvideoencodingparameters();
@@ -74,7 +75,8 @@
     if (audiostreammask & (1u<<i)) {
       int astr=audiostream(i);
       st[astr].stream_index=id;
-      s=st[astr].avstr=av_new_stream(avfc,id++);
+      s=st[astr].avstr=avformat_new_stream(avfc,NULL);
+      s->id = id++;
       strpres[astr]=true;
       if (s->codec)
         av_free(s->codec);
