--- makefile.in.orig	2011-03-16 23:04:06.364941975 +1100
+++ makefile.in	2011-03-16 23:08:32.197955984 +1100
@@ -29,9 +29,9 @@
 pkgdatadir = $(datadir)/dvbcut
 
 installdirs = $(DESTDIR)$(man1dir) \
-	$(DESTDIR)$(pkgdatadir)/icons \
-	$(DESTDIR)/usr/share/applications \
-	$(DESTDIR)/usr/share/mime/packages
+	$(DESTDIR)$(datadir)/icons/hicolor/scalable/apps \
+	$(DESTDIR)$(datadir)/applications \
+	$(DESTDIR)$(datadir)/mime/packages
 
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
@@ -46,10 +46,9 @@
 dep-local:
 install-local: $(installdirs) dvbcut.1 dvbcut.desktop dvbcut.xml
 	$(INSTALL_DATA) dvbcut.1 $(DESTDIR)$(man1dir)/dvbcut.1
-	$(INSTALL_DATA) dvbcut.desktop $(DESTDIR)/usr/share/applications
-	$(INSTALL_DATA) dvbcut.svg $(DESTDIR)$(pkgdatadir)/icons
-	$(INSTALL_DATA) dvbcut.xml $(DESTDIR)/usr/share/mime/packages
-	-update-mime-database $(DESTDIR)/usr/share/mime
+	$(INSTALL_DATA) dvbcut.svg $(DESTDIR)$(datadir)/icons/hicolor/scalable/apps
+	$(INSTALL_DATA) dvbcut.desktop $(DESTDIR)$(datadir)/applications
+	$(INSTALL_DATA) dvbcut.xml $(DESTDIR)$(datadir)/mime/packages
 
 $(installdirs):
 	$(SHELL) $(top_srcdir)/mkinstalldirs $@
